xbmc/guilib/StereoscopicsManager.cpp

```diff
@@ -509,6 +509,8 @@ void CStereoscopicsManager::OnPlaybackStarted(void)
   STEREOSCOPIC_PLAYBACK_MODE playbackMode = (STEREOSCOPIC_PLAYBACK_MODE) CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_STEREOSCOPICPLAYBACKMODE);
   RENDER_STEREO_MODE mode = GetStereoMode();
 
+  g_infoManager.UpdateAVInfo();
```

Comment:
well, it basically does two things. First task is to apply the users stereoscopic preferences to video playback (and once/if ever supported then also to pictures). I tried to do this as least intrusive as possible and thus used messages/signals to achieve this. It should also be the only entity that handles the GUI settings, so (at least in my vision) no other component should ever directly access stereoscopic related settings via the settings manager but instead always ask the manager. Once the GUI setting is being changed, the manager informs the required components about this change (curently IIRC only g_graphicsContext)

Second task is to act as connector for stereoscopic interaction between components. At least the connection from interfaces and GUI to core components. It also contains some utility functions for mode conversions (GUI enums to skiin/interfaces) that probably should have been moved to a utility class.